### ISSUE TITLE
Why `sigmoid_()` do not take effect when applied to a indexed tensor? 

### ISSUE THREAD

**Issue by songyuc** [pytorch/pytorch#80945]
### 🐛 Describe the bug

`sigmoid_()` do not take effect when applied to a indexed tensor.

```python
a = torch.tensor([[1, 0, 3.1],
                  [4.1, 5.1, 6.1]])
b = a.log().abs()
print(b)
mask = b.isinf()
print(b[mask].sigmoid_())
print(b)
>>>
tensor([[0.0000,    inf, 1.1314],
        [1.4110, 1.6292, 1.8083]])
tensor([1.])
tensor([[0.0000,    inf, 1.1314],
        [1.4110, 1.6292, 1.8083]])
```

### Versions

Collecting environment information...
PyTorch version: 1.11.0+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.5
Libc version: glibc-2.26

Python version: 3.7.13 (default, Apr 24 2022, 01:04:09)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.188+-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: False
CUDA runtime version: 11.1.105
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.0.5
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.6
[pip3] torch==1.11.0+cu113
[pip3] torchaudio==0.11.0+cu113
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.12.0
[pip3] torchvision==0.12.0+cu113
[conda] Could not collect

**Comment by ngimel** [comment]
This is expected behavior, `b[mask]` returns a new intermediate tensor to which `sigmoid_` is applied inplace, and intermediate disappears at the end of the line.

**Comment by songyuc** [comment]
@ngimel, but why **b[mask]=10** will do an inplace assignment. It seems there is a **inconsistency** here!

**Comment by ngimel** [comment]
In some limited scenarios augmented assignment works https://www.pythonlikeyoumeanit.com/Module3_IntroducingNumpy/AdvancedIndexing.html#In-Place-&-Augmented-Assignments-via-Advanced-Indexing, but generally you should not rely on it because you are creating a new intermediate tensor.
